lib/recart-nacelle-plugin.js

```diff
@@ -1,7 +1,34 @@
 const getMetafieldsForCheckout = require('./getMetafieldsForCheckout')
+const { saveCart } = require('./cart')
 
-const initRecartPlugin = async (ctx, inject) => {
+const delay = async milliseconds => {
+  return new Promise(resolve =>
+    setTimeout(() => resolve(true), milliseconds)
+  )
+}
+
+const initRecartPlugin = async (ctx) => {
   console.log('Recart Nacelle plugin started')
+
+  let timoutCounter = 0
+  while (!window._recart || !window._recart.isReady()) {
+    if (timoutCounter > 10) throw new Error('Recart is not initialized.')
```

Comment:
🤔  should we really throw an error here? what is your line of thinking?

what about just logging an error and returning?